### PR TITLE
Sanlouise 14012 hide profile error

### PR DESCRIPTION
**Pull request by sanlouise**
## Description
We should not show the global profile error when the call to `/service_history` fails and the user is a non-veteran. The call fails expectedly with a 403, so this should not be treated as an error.

## Testing done
Works locally, added unit tests.

## Acceptance criteria
- [x] Hide error when the call to `/service_history` fails and the user is a non-veteran

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile-2/tests/components/Profile.data-unavailable-error.unit.spec.js [hunk 1]
```diff
@@ -93,6 +93,32 @@ async function errorAppearsOnAllPages(
...
+  const initialState = createBasicInitialState();
+  initialState.user.profile.veteranStatus = null;
+
+  const view = render(<Profile2Router isLOA3 isInMVI />, {
+    initialState,
+  });
+  const spinner = view.queryByRole('progressbar');
+  await waitForElementToBeRemoved(spinner);
+
+  const alert = await view.queryByTestId(ALERT_ID);
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/tests/components/Profile.data-unavailable-error.unit.spec.js:113]: should be no need to `await` on `queryBy...`. `await` is just used for `findBy` queries, I believe.

File: src/applications/personalization/profile-2/tests/components/Profile.data-unavailable-error.unit.spec.js [hunk 2]
```diff
@@ -93,6 +93,32 @@ async function errorAppearsOnAllPages(
...
+  expect(alert).not.to.exist;
+
+  pageNames.forEach(pageName => {
+    userEvent.click(view.getByRole('link', { name: pageName }));
+    expect(alert).to.be.null;
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/tests/components/Profile.data-unavailable-error.unit.spec.js:118]: I think that for this `expect` to do what you want you'd really need to rerun the query:
`expect(view.queryByTextId(ALERT_ID)).to.not.exist;`

**Review by erikphansen — APPROVED** [review]

File: src/applications/personalization/profile-2/tests/components/Profile.data-unavailable-error.unit.spec.js (hunk 1 above)
Comment (sanlouise) [reply src/applications/personalization/profile-2/tests/components/Profile.data-unavailable-error.unit.spec.js]: you were right! updated! 🎉

File: src/applications/personalization/profile-2/tests/components/Profile.data-unavailable-error.unit.spec.js (hunk 2 above)
Comment (sanlouise) [reply src/applications/personalization/profile-2/tests/components/Profile.data-unavailable-error.unit.spec.js]: Made the update, thanks for the suggestion!

**Pull request merged**